### PR TITLE
MEN-3510 use attribute retrieval endpoint

### PR DESCRIPTION
**Pull request by mzedel**
this relies on the changes in #1002 to line up the app initialisation steps
main change is in f76bc0dee687b55af08780f263b520172cfd54a5

### CODE REVIEW

File: src/js/actions/deviceActions.js [hunk 1]
```diff
@@ -549,6 +536,24 @@ export const getAllDevicesByStatus = status => (dispatch, getState) => {
   return getAllDevices();
 };
 
+export const getDeviceAttributes = () => dispatch =>
+  GeneralApi.get(`${inventoryApiUrlV2}/filters/attributes`).then(({ data = [] }) => {
```
Comment (tranchitella) [inline src/js/actions/deviceActions.js:540]: We return up to 500 attributes here, not sure you need a limit in the UI... maybe not,

File: src/js/helpers.js [hunk 2]
```diff
@@ -318,35 +318,6 @@ export const mapDeviceAttributes = (attributes = []) =>
...
-  // sort in reverse order, to have most common attribute at the top of the select
-  const inventoryAttributes = Object.entries(availableAttributes.attributes)
-    .sort((a, b) => b[1] - a[1])
-    .map(a => a[0]);
-  const identityAttributes = Object.entries(availableAttributes.identity_data)
-    .sort((a, b) => b[1] - a[1])
-    .map(a => a[0]);
-  return { identityAttributes, inventoryAttributes };
-};
-
```
Comment (tranchitella) [inline src/js/helpers.js:349]: So nice to get rid of this custom logic in the UI, yay!

**Review by tranchitella — APPROVED** [review]
Great job!

File: src/js/actions/deviceActions.js (hunk 1 above)
Comment (mzedel) [reply src/js/actions/deviceActions.js:540]: a valid point indeed - we once had a limit, but soon after got feedback that there are some attributes missing 🙈... while the list of identity attributes is still limited, the filters might show all... the component we use there is definitely capable of a few hundred items though so we should be good 👌

**Pull request merged**